tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -605,13 +605,28 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
         return false;
     }
 
-    // TODO(philipphack): Consider enabling epilogue fusions and the addition of
-    // a matrix bias for FP8 GEMMs.
-    Literal c_literal = LiteralUtil::Zero(c_type);
-    HloInstruction *c = instr->AddInstruction(
-        HloInstruction::CreateConstant(c_literal.Clone()));
-    HloInstruction *c_bcast = instr->AddInstruction(
-        HloInstruction::CreateBroadcast(instr->shape(), c, {}));
+    // Fuse the possible addition of a matrix bias here to enable the subsequent
+    // fusion of the scaling and conversion of D into the Custom Call.
+    HloInstruction *c = nullptr;
+    if (instr->user_count() == 1 &&
+        instr->users()[0]->opcode() == HloOpcode::kAdd) {
+      HloInstruction *add = instr->users()[0];
+      HloInstruction *bias = add->mutable_operand(!add->operand_index(instr));
+      if (bias->opcode() != HloOpcode::kBroadcast) {
+        c = bias;
+        gemm_backend_config.set_beta(1.0);
+        TF_RETURN_IF_ERROR(ReplaceInstruction(add, instr));
```

Comment:
Why not handle the bias logic in handleAdd, [as is already done in the non-FP8 case](https://github.com/tensorflow/tensorflow/blob/1f747f5b52003896c5fe6723ff95f84f00def94a/tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc#L378)? handleAdd can fuse a vector bias addition, which seems beneficial.

Reply:
The implementation is here to facilitate the subsequent fusion of the quantization of the result (if present and compatible with the bias fusion). This assumes that the bias is added before these operations, i.e. the bias is prescribed in the dequantized type. HandleAdd matches both matrix and vector bias additions but calls separate functions to do so.

Reply:
Why would handling the bias logic here instead of in HandleAdd make the subsequent fusion of the result quantization easier? If the bias logic was handled in HandleAdd, we could still quantize the output HandleConvert after fusing the bias in HandleAdd.

Reply:
Your point about the FP8 Custom call always having a third "C" operand makes sense. Can't the aliasing logic also apply to FP8 if the we don't quantize the output?

Anyway we can always fix this later and the fact there is a third "C" operand would make changing this PR a bit tricky, so I'll approve and merge for now.

Reply:
I asked internally about the possibility of aliasing in this case. If we want to know at the time of fusing the bias if the output will be quantized, we would still have to do some form of top-down graph inspection.